### DISCUSSION TITLE
findFirstRecordByFilter with Admin credentials

### DISCUSSION THREAD

**Discussion by charbs-io** [pocketbase/pocketbase#3107] [Q&A]
Hi, 

I'm experimenting with the JSVM, and i'm using `$app.dao().findFirstRecordByFilter()` to find a certain record. 

API rules set on the collection forbid the authenticated user from accessing a given record - that is expected.

My question is, am I able to "authenticate" as an admin on `routerAdd` so i can find this record and do what I want with it? 

The context:
User A intends to invite john@smith.com to their workspace.
I am using a custom route in pb_hooks to register the route and ultimately create records etc.
The one check i have to make first though, is whether john@smith.com exists on the db to begin with, because his experience (i.e emails i would send etc) would be different than someone not on the db.

Edit: I know I can potentially use `findAuthRecordByEmail`, but the collection i am searching for is not an auth collection, it's a `workspaceUser` one where it holds relations for `users` and `workspaces`.

Thanks!

**Comment by ganigeorgiev** [comment]
>  am I able to "authenticate" as an admin on routerAdd so i can find this record and do what I want with it?

I'm not entirely sure I follow. Inside the custom route handler itself there are **no API rules applied by default, aka. you are interacting directly with the database**. If you want only an admin to be able to access the route you need to register the `$apis.requireAdminAuth()` middleware (see [Middlewares](https://pocketbase.io/docs/js-routing/#middlewares)).

_If the above is not what you are looking for, could you elaborate or provide a pseudo-code of what you are trying to do?_

**Comment by charbs-io** [comment]
Hey @ganigeorgiev - thanks for your reply.

Ohhh - that makes sense. I'm so sorry. This is my own mistake I think.

```
// Check if the user exists in the workspace
const workspaceUserRecord = $app
      .dao()
      ?.findFirstRecordByFilter(
        "workspaceUsers",
        `user = '${userRecord.id}' && workspace = '${workspaceId}'`
      );
```

This was returning `sql: no rows in result set` in the console - I assumed this was occurring because technically the authenticated user does not have access to the record (because they shouldn't), but actually this is just happening because it actually cannot find that user in the workspace...

I suppose a follow up question would be - how do i catch up that error? Because it shouldn't kill the rest of the execution (currently the SDK returns 404 because of it)

**Reply by ganigeorgiev** [reply]
You can try to wrap the thrown exception with `try/catch`.

_One side-note: `findFirstRecordByFilter()` is not safe to be used with untrusted user input. If `workspaceId` is not trusted, a better approach would be to construct a `RecordQuery()` on your own, aka.:_

```js
const workspaceUserRecord = new Record();

$app.dao().recordQuery("workspaceUsers")
    .andWhere($dbx.hashExp({
        "user":      userRecord.id,
        "workspace": workspaceId,
    }))
    .limit(1)
    .one(workspaceUserRecord)

```
